mex/editor/search/state.py

```diff
@@ -142,11 +146,28 @@ def all_fields_for_entity_types(self) -> list[str]:
         if len(selected_entity_types) == 0:
             selected_entity_types = [item[0] for item in self.entity_types.items()]
 
-        fields_by_type = [
-            REFERENCE_FIELDS_BY_CLASS_NAME[ensure_prefix(entity_type, "Extracted")]
+        fields_with_type = [
+            [
+                {
+                    "value": field,
+                    "label": locale_service.get_field_label(
+                        self.current_locale, entity_type, field
+                    ),
+                }
```

Comment:
could you create a model for this dict - and then indicate in `CustomHighLevelSelect` that it is expecting its menu items to have this model as a type?